[BUG] Import of password protected file fails

### Description

![image](https://github.com/user-attachments/assets/ad9407ec-33db-45eb-a15e-d1e6aff9358b)
This had been possible in the past without giving a password. Password request then occured when document is displayed during further processing but not at import time.

### Steps to reproduce

Import password protected file.

### Webserver logs

```bash
[2024-10-29 13:30:41,163] [INFO] [paperless.consumer] Consuming Lohnscheine_202410.pdf

[2024-10-29 13:30:41,165] [DEBUG] [paperless.consumer] Detected mime type: application/pdf

[2024-10-29 13:30:41,171] [DEBUG] [paperless.consumer] Parser: RasterisedDocumentParser

[2024-10-29 13:30:41,175] [DEBUG] [paperless.consumer] Parsing Lohnscheine_202410.pdf...

[2024-10-29 13:30:41,183] [INFO] [paperless.parsing.tesseract] pdftotext exited 1

[2024-10-29 13:30:41,183] [WARNING] [paperless.parsing.tesseract] Error while getting text from PDF document with pdftotext

Traceback (most recent call last):

  File "/usr/src/paperless/src/paperless_tesseract/parsers.py", line 181, in extract_text

    run_subprocess(

  File "/usr/src/paperless/src/documents/utils.py", line 108, in run_subprocess

    completed_proc.check_returncode()

  File "/usr/local/lib/python3.12/subprocess.py", line 502, in check_returncode

    raise CalledProcessError(self.returncode, self.args, self.stdout,

subprocess.CalledProcessError: Command '['pdftotext', '-q', '-layout', '-enc', 'UTF-8', PosixPath('/tmp/paperless/paperless-ngxos9aq0fc/Lohnscheine_202410.pdf'), '/tmp/paperless/paperless-lleveczh/tmphllzqp1l']' returned non-zero exit status 1.

[2024-10-29 13:30:41,297] [DEBUG] [paperless.parsing.tesseract] Calling OCRmyPDF with args: {'input_file': PosixPath('/tmp/paperless/paperless-ngxos9aq0fc/Lohnscheine_202410.pdf'), 'output_file': PosixPath('/tmp/paperless/paperless-lleveczh/archive.pdf'), 'use_threads': True, 'jobs': 2, 'language': 'deu+eng', 'output_type': 'pdfa', 'progress_bar': False, 'color_conversion_strategy': 'RGB', 'skip_text': True, 'clean': True, 'deskew': True, 'rotate_pages': True, 'rotate_pages_threshold': 12.0, 'sidecar': PosixPath('/tmp/paperless/paperless-lleveczh/sidecar.txt')}

[2024-10-29 13:30:41,426] [WARNING] [paperless.parsing.tesseract] This file is encrypted, OCR is impossible. Using any text present in the original file.

[2024-10-29 13:30:41,426] [WARNING] [paperless.parsing.tesseract] No text was found in /tmp/paperless/paperless-ngxos9aq0fc/Lohnscheine_202410.pdf, the content will be empty.

[2024-10-29 13:30:41,427] [DEBUG] [paperless.consumer] Generating thumbnail for Lohnscheine_202410.pdf...

[2024-10-29 13:30:41,430] [DEBUG] [paperless.parsing] Execute: convert -density 300 -scale 500x5000> -alpha remove -strip -auto-orient -define pdf:use-cropbox=true /tmp/paperless/paperless-ngxos9aq0fc/Lohnscheine_202410.pdf[0] /tmp/paperless/paperless-lleveczh/convert.webp

[2024-10-29 13:30:41,528] [INFO] [paperless.parsing] convert exited 1

[2024-10-29 13:30:41,528] [INFO] [paperless.parsing] convert stderr:

[2024-10-29 13:30:41,528] [WARNING] [paperless.parsing] GPL Ghostscript 10.03.1:

[2024-10-29 13:30:41,529] [WARNING] [paperless.parsing]    **** This file requires a password for access.

[2024-10-29 13:30:41,529] [WARNING] [paperless.parsing]    **** Error: Couldn't initialise file.

[2024-10-29 13:30:41,529] [WARNING] [paperless.parsing]                Output may be incorrect.

[2024-10-29 13:30:41,529] [WARNING] [paperless.parsing]

[2024-10-29 13:30:41,530] [WARNING] [paperless.parsing] Requested FirstPage is greater than the number of pages in the file: 0

[2024-10-29 13:30:41,530] [WARNING] [paperless.parsing]    No pages will be processed (FirstPage > LastPage).

[2024-10-29 13:30:41,530] [WARNING] [paperless.parsing] convert-im6.q16: no images defined `/tmp/paperless/paperless-lleveczh/convert.webp' @ error/convert.c/ConvertImageCommand/3229.

[2024-10-29 13:30:41,531] [ERROR] [paperless.parsing] Unable to make thumbnail with convert: Convert failed at ['convert', '-density', '300', '-scale', '500x5000>', '-alpha', 'remove', '-strip', '-auto-orient', '-define', 'pdf:use-cropbox=true', '/tmp/paperless/paperless-ngxos9aq0fc/Lohnscheine_202410.pdf[0]', '/tmp/paperless/paperless-lleveczh/convert.webp']

[2024-10-29 13:30:41,531] [WARNING] [paperless.parsing] Thumbnail generation with ImageMagick failed, falling back to ghostscript. Check your /etc/ImageMagick-x/policy.xml!

[2024-10-29 13:30:41,587] [INFO] [paperless.parsing] gs exited 0

[2024-10-29 13:30:41,587] [INFO] [paperless.parsing] gs stdout:

[2024-10-29 13:30:41,587] [INFO] [paperless.parsing] **** Error: Couldn't initialise file.

[2024-10-29 13:30:41,588] [INFO] [paperless.parsing]                Output may be incorrect.

[2024-10-29 13:30:41,588] [INFO] [paperless.parsing]    No pages will be processed (FirstPage > LastPage).

[2024-10-29 13:30:41,588] [INFO] [paperless.parsing] gs stderr:

[2024-10-29 13:30:41,588] [WARNING] [paperless.parsing] GPL Ghostscript 10.03.1:

[2024-10-29 13:30:41,589] [WARNING] [paperless.parsing]    **** This file requires a password for access.

[2024-10-29 13:30:41,589] [DEBUG] [paperless.parsing] Execute: convert -density 300 -scale 500x5000> -alpha remove -strip -auto-orient /tmp/paperless/paperless-lleveczh/gs_out.png /tmp/paperless/paperless-lleveczh/convert_gs.webp

[2024-10-29 13:30:41,596] [INFO] [paperless.parsing] convert exited 1

[2024-10-29 13:30:41,597] [INFO] [paperless.parsing] convert stderr:

[2024-10-29 13:30:41,597] [WARNING] [paperless.parsing] convert-im6.q16: unable to open image `/tmp/paperless/paperless-lleveczh/gs_out.png': No such file or directory @ error/blob.c/OpenBlob/2924.

[2024-10-29 13:30:41,597] [WARNING] [paperless.parsing] convert-im6.q16: no images defined `/tmp/paperless/paperless-lleveczh/convert_gs.webp' @ error/convert.c/ConvertImageCommand/3229.

[2024-10-29 13:30:41,597] [ERROR] [paperless.parsing] Unable to make thumbnail with Ghostscript: Convert failed at ['convert', '-density', '300', '-scale', '500x5000>', '-alpha', 'remove', '-strip', '-auto-orient', '/tmp/paperless/paperless-lleveczh/gs_out.png', '/tmp/paperless/paperless-lleveczh/convert_gs.webp']

[2024-10-29 13:30:41,602] [DEBUG] [paperless.parsing.tesseract] Deleting directory /tmp/paperless/paperless-lleveczh

[2024-10-29 13:30:41,604] [ERROR] [paperless.consumer] Unexpected error while consuming document Lohnscheine_202410.pdf: /tmp/paperless/paperless-ngxos9aq0fc/Lohnscheine_202410.pdf: invalid password

Traceback (most recent call last):

  File "/usr/local/lib/python3.12/site-packages/asgiref/sync.py", line 327, in main_wrap

    raise exc_info[1]

  File "/usr/src/paperless/src/documents/consumer.py", line 502, in run

    page_count = document_parser.get_page_count(self.working_copy, mime_type)

                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

  File "/usr/src/paperless/src/paperless_tesseract/parsers.py", line 48, in get_page_count

    with pikepdf.Pdf.open(document_path) as pdf:

         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

  File "/usr/local/lib/python3.12/site-packages/pikepdf/_methods.py", line 399, in open

    pdf = Pdf._open(

          ^^^^^^^^^^

pikepdf._core.PasswordError: /tmp/paperless/paperless-ngxos9aq0fc/Lohnscheine_202410.pdf: invalid password

[2024-10-29 13:30:41,606] [ERROR] [paperless.tasks] ConsumeTaskPlugin failed: Lohnscheine_202410.pdf: Unexpected error while consuming document Lohnscheine_202410.pdf: /tmp/paperless/paperless-ngxos9aq0fc/Lohnscheine_202410.pdf: invalid password

Traceback (most recent call last):

  File "/usr/local/lib/python3.12/site-packages/asgiref/sync.py", line 327, in main_wrap

    raise exc_info[1]

  File "/usr/src/paperless/src/documents/consumer.py", line 502, in run

    page_count = document_parser.get_page_count(self.working_copy, mime_type)

                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

  File "/usr/src/paperless/src/paperless_tesseract/parsers.py", line 48, in get_page_count

    with pikepdf.Pdf.open(document_path) as pdf:

         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

  File "/usr/local/lib/python3.12/site-packages/pikepdf/_methods.py", line 399, in open

    pdf = Pdf._open(

          ^^^^^^^^^^

pikepdf._core.PasswordError: /tmp/paperless/paperless-ngxos9aq0fc/Lohnscheine_202410.pdf: invalid password


The above exception was the direct cause of the following exception:


Traceback (most recent call last):

  File "/usr/src/paperless/src/documents/tasks.py", line 148, in consume_file

    msg = plugin.run()

          ^^^^^^^^^^^^

  File "/usr/src/paperless/src/documents/consumer.py", line 518, in run

    self._fail(

  File "/usr/src/paperless/src/documents/consumer.py", line 151, in _fail

    raise ConsumerError(f"{self.filename}: {log_message or message}") from exception

documents.consumer.ConsumerError: Lohnscheine_202410.pdf: Unexpected error while consuming document Lohnscheine_202410.pdf: /tmp/paperless/paperless-ngxos9aq0fc/Lohnscheine_202410.pdf: invalid password
```


### Browser logs

_No response_

### Paperless-ngx version

2.13.1

### Host OS

Ubuntu

### Installation method

Docker - official image

### System status

```json
{
    "pngx_version": "2.13.1",
    "server_os": "Linux-5.15.0-124-generic-x86_64-with-glibc2.36",
    "install_type": "docker",
    "storage": {
        "total": 105089261568,
        "available": 56434229248
    },
    "database": {
        "type": "postgresql",
        "url": "paperless",
        "status": "OK",
        "error": null,
        "migration_status": {
            "latest_migration": "documents.1056_customfieldinstance_deleted_at_and_more",
            "unapplied_migrations": []
        }
    },
    "tasks": {
        "redis_url": "redis://broker:6379",
        "redis_status": "OK",
        "redis_error": null,
        "celery_status": "OK",
        "index_status": "OK",
        "index_last_modified": "2024-10-29T00:00:04.044720+01:00",
        "index_error": null,
        "classifier_status": "OK",
        "classifier_last_trained": "2024-10-29T12:05:00.577905Z",
        "classifier_error": null
    }
}
```


### Browser

Edge

### Configuration changes

_No response_

### Please confirm the following

- [X] I believe this issue is a bug that affects all users of Paperless-ngx, not something specific to my installation.
- [X] This issue is not about the OCR or archive creation of a specific file(s). Otherwise, please see above regarding OCR tools.
- [X] I have already searched for relevant existing issues and discussions before opening this report.
- [X] I have updated the title field above with a concise description.

--- a/src/paperless_tesseract/parsers.py
+++ b/src/paperless_tesseract/parsers.py
@@ -43,10 +43,15 @@ class RasterisedDocumentParser(DocumentParser):
     def get_page_count(self, document_path, mime_type):
         page_count = None
         if mime_type == "application/pdf":
-            import pikepdf
+            try:
+                import pikepdf
 
-            with pikepdf.Pdf.open(document_path) as pdf:
-                page_count = len(pdf.pages)
+                with pikepdf.Pdf.open(document_path) as pdf:
+                    page_count = len(pdf.pages)
+            except Exception as e:
+                self.log.warning(
+                    f"Unable to determine PDF page count {document_path}: {e}",
+                )
         return page_count
 
     def extract_metadata(self, document_path, mime_type):

--- a/src/paperless_tesseract/tests/test_parser.py
+++ b/src/paperless_tesseract/tests/test_parser.py
@@ -81,6 +81,24 @@ class TestParser(DirectoriesMixin, FileSystemAssertsMixin, TestCase):
         )
         self.assertEqual(page_count, 6)
 
+    def test_get_page_count_password_protected(self):
+        """
+        GIVEN:
+            - Password protected PDF file
+        WHEN:
+            - The number of pages is requested
+        THEN:
+            - The method returns None
+        """
+        parser = RasterisedDocumentParser(uuid.uuid4())
+        with self.assertLogs("paperless.parsing.tesseract", level="WARNING") as cm:
+            page_count = parser.get_page_count(
+                os.path.join(self.SAMPLE_FILES, "password-protected.pdf"),
+                "application/pdf",
+            )
+            self.assertEqual(page_count, None)
+            self.assertIn("Unable to determine PDF page count", cm.output[0])
+
     def test_thumbnail(self):
         parser = RasterisedDocumentParser(uuid.uuid4())
         thumb = parser.get_thumbnail(

